alternator/executor.cc

```diff
@@ -3616,10 +3616,16 @@ static rjson::value encode_paging_state(const schema& schema, const service::pag
             ++exploded_ck_it;
         }
     }
-    rjson::add_with_string_name(last_evaluated_key, scylla_paging_region, rjson::empty_object());
-    rjson::add(last_evaluated_key[scylla_paging_region.data()], "S", rjson::from_string(to_string(pos.region())));
-    rjson::add_with_string_name(last_evaluated_key, scylla_paging_weight, rjson::empty_object());
-    rjson::add(last_evaluated_key[scylla_paging_weight.data()], "N", static_cast<int>(pos.get_bound_weight()));
+    // The paging-state cookie only needs full position-in-partition support for
+    // tables that can have range-tombstones. As alternator itself doesn't allow
+    // these, we only need the position-in-partition support when reading CQL
+    // tables through alternator. Note that this is only supported for system tables.
+    if (!executor::is_alternator_keyspace(schema.ks_name())) {
```

Comment:
A thought (I don't feel strongly about it, I guess your implementation is good too):

This means that for non-Alternator tables (i.e., system tables), we always write those extra things in LastEvaluatedKey, even if usually unnecessary. I wonder if wouldn't have been simpler to replace this if with two ifs (with faster conditions) - don't print the weight if it is "after", and don't print the region if it is "clustering".

Reply:
I like this, it would lead to an even smaller patch.

Reply:
> sounds even better, and more technically correct - if somehow we ever introduce range tombstones to alternator, the code will remain correct and would automatically pick up our magic extension system with weights and regions

However, note that you can only do this on the LastEvaluatedKey write stage! In the ExclusiveKeyStart reading stage you still need to check if it's an Alternator table or not, because if you don't, you'll bring back the (esoteric) problem of forbidden key column names in Alternator.

Reply:
I don't think that is a good idea: the checks have to be symmetric in my opinion. So I guess we go back to v1 then?